### DISCUSSION TITLE
Preferred App Architecture

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#28] [Off topic]
Hi, do you use a defined App Architecture?
E.G. https://codewithandrea.com/articles/flutter-app-architecture-riverpod-introduction/#riverpod-is-not-very-opinionated

**Comment by GregoryConrad** [comment]
I tend to find clear cut layers more of a mental exercise rather than a productivity or maintainability boost. If you tend to follow one approach to a T, you’ll likely spend a ton of time bikeshedding on where to actually put something or making connections between layers instead of actually developing.

That being said, you have the flexibility to structure your code however you want. I tend to like feature-oriented development; i.e., keep the code corresponding to some feature/page together so it’s all easy to find and modify in the future. If you’d like, you can also make some utility classes/capsules and abstractions for your backend so you can swap those implementations out in the future, but not required depending on the scope of the project.

**Reply by busslina** [reply]
Cool, I will continue to use my own way (a sort of featured oriented)
